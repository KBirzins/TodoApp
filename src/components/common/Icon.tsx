import React from 'react';
import {TouchableWithoutFeedback} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

interface IIcon {
  size?: number;
  color?: string;
  name: string;
  onPress?(): any;
}

const StyledIcon = ({size = 25, color = '#211f30', name, onPress}: IIcon) => {
  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <Icon size={size} color={color} name={name} />
    </TouchableWithoutFeedback>
  );
};

export default StyledIcon;
