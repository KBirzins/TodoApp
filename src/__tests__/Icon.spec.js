import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Icon from '../components/common/Icon';

const NAME = 'edit';

describe('Icon', () => {
  it('renders correctly with correct properties', () => {
    const component = renderer.create(<Icon name={NAME} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    const root = component.root;

    expect(root.findByType(Icon).props.name).toBe(NAME);
  });
});
