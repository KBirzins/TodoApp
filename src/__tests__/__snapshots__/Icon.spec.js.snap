// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly with correct properties 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "color": "#211f30",
        "fontSize": 25,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  
</Text>
`;
